### PR TITLE
Added vote notification functionality

### LINKED ISSUE #29: Author not getting notification of new follower.
When anybody votes up a story so the user should get a notification that who voted up their story.

### PR DESCRIPTION
**Pull request by codetheorem**
Signed-off-by: Hrishikesh Agarwal <hrishikeshagarwalv@gmail.com>
Fixes #29 

Added a feature that when a user votes up a story the author of the story will get a notification that who voted up the story.

### CODE REVIEW

File: api/user-story/models/user-story.js [hunk 1]
```diff
@@ -23,6 +23,23 @@ module.exports = {
...
       }
+
+      catch(e) {}
+    },
+    async beforeUpdate(params,data) {
+      try {
+              const story=await strapi.query('user-story').findOne({id: params.id})
+              const userId= data.followers[data.followers.length-1]
+              if(story?.followers?.length < data?.followers?.length && story.author.id !== userId) {
+              const user= await strapi.query('user', 'users-permissions').findOne({id: userId})
```
Comment (cyntss) [inline api/user-story/models/user-story.js]: Can you take a look at your indentation and make sure that everything is correct? we use 2 spaces. thanks

**Review by cyntss — CHANGES_REQUESTED** [review]

File: api/user-story/models/user-story.js [hunk 2]
```diff
@@ -7,9 +7,9 @@
 
 module.exports = {
   lifecycles: {
-    async afterUpdate(result) {
-      try {
-        if(result.user_story_status.Status === 'Deployed') {
+    async afterUpdate(result){
+      try{
+          if(result.user_story_status.Status === 'Deployed') {
```
Comment (cyntss) [inline api/user-story/models/user-story.js]: here there are 2 additional spaces before the `if`

File: api/user-story/models/user-story.js [hunk 3]
```diff
@@ -23,6 +23,23 @@ module.exports = {
...
+           const userId= data.followers[data.followers.length-1]
+           if(story?.followers?.length < data?.followers?.length && story.author.id !== userId) {
+           const user= await strapi.query('user', 'users-permissions').findOne({id: userId})
+           await strapi.services['user-story-notification'].create({
+            message: `${user.username} voted up your story`,
+            users: [story.author],
+            date: new Date(),
+            link: `story/${story.id}`
+           })
+          }
```
Comment (cyntss) [inline api/user-story/models/user-story.js]: From line 31 to line 41 the indentation is not correct. I suggest that you configure your IDE to automatically format your indentation with 2 spaces. Visual Code Studio has this option for Formatting.

It's a very good practice to have consistent indentation and formatting, and we're very picky about it at EOS, so let's make sure this file is set correctly

**Review by cyntss — CHANGES_REQUESTED** [review]

File: api/user-story/models/user-story.js (hunk 3 above)
Comment (codetheorem) [reply api/user-story/models/user-story.js]: Done.

File: api/user-story/models/user-story.js (hunk 2 above)
Comment (codetheorem) [reply api/user-story/models/user-story.js]: Done

File: api/user-story/models/user-story.js (hunk 1 above)
Comment (codetheorem) [reply api/user-story/models/user-story.js]: Done

File: api/user-story/models/user-story.js [hunk 4]
```diff
@@ -22,35 +22,38 @@ module.exports = {
...
+          })
           data.user_story_status = status.id
         } catch (e) {
           console.log('Unable to fetch "Under consideration" status')
         }
       }
     }
   }
-};
+};
```
Comment (cyntss) [inline api/user-story/models/user-story.js]: Please add an empty line at the end of the file

**Review by cyntss — CHANGES_REQUESTED** [review]
Please fix the last comment added and we can merge this

File: api/user-story/models/user-story.js (hunk 4 above)
Comment (codetheorem) [reply api/user-story/models/user-story.js]: Done.

**Review by cyntss — APPROVED** [review]
lgtm

**Pull request merged**